### PR TITLE
Updating mock_client_transport to work with None default.

### PR DESCRIPTION
**Pull request by err508**
Fixing missing source address in request_paths method.

### CODE REVIEW

File: raiden_libs/test/mocks/client.py [hunk 1]
```diff
@@ -368,3 +366,8 @@ def get_channel_participant_info(
             field: channel_info[return_fields.index(field)]
             for field in return_fields
         }
+
+    def set_callback(self):
+        """To be called if a transport is used."""
+        assert self.transport is not None
+        self.transport.add_message_callback(self.on_message_event)
```
Comment (palango) [inline raiden_libs/test/mocks/client.py]: Why not do this in the constructor as before?

File: raiden_libs/test/mocks/client.py [hunk 2]
```diff
@@ -78,9 +78,6 @@ def __init__(
         self.web3 = self.contract.web3
         self.transport = transport
         self.paths_and_fees = None
-        if self.transport is not None:
-            self.transport.start()  # type: ignore
```
Comment (palango) [inline raiden_libs/test/mocks/client.py]: Is the call of `start`no needed any more?

File: raiden_libs/test/mocks/client.py (hunk 1 above)
Comment (err508) [reply raiden_libs/test/mocks/client.py]: This wasn't working, the `self.transport.add_message_callback` was producing an error with `transport = None`.

File: raiden_libs/test/mocks/client.py (hunk 2 above)
Comment (err508) [reply raiden_libs/test/mocks/client.py]: Nope, is working without it. Same with `pfs.transport`.

File: raiden_libs/test/mocks/client.py (hunk 1 above)
Comment (palango) [reply raiden_libs/test/mocks/client.py]: But that should have never been called, right?

Comment (err508) [reply raiden_libs/test/mocks/client.py]: Yupp, was never called. I wanted to make it clear, that this is optional for the mocked_client, that's the reason for the additional method.

**Pull request merged**